posthog/api/decide.py

```diff
@@ -94,7 +94,14 @@ def get_base_config(token: str, team: Team, request: HttpRequest, skip_db: bool
 
     REMOTE_CONFIG_CACHE_COUNTER.labels(result=use_remote_config).inc()
 
-    surveys_opt_in = get_surveys_opt_in(team) and get_surveys_count(team) > 0
+    # errors mean the database is unavailable, rely on team setting in this case
+    surveys_opt_in = get_surveys_opt_in(team)
+    if surveys_opt_in and not skip_db:
+        try:
+            with execute_with_timeout(200):
+                surveys_opt_in = get_surveys_count(team) > 0
+        except Exception:
+            pass
```

Comment:
I roughly copied this from what we have for site apps. @benjackwhite we obviously fetch the team and / or user at other points of decide. Is it safe to assume that those will likely be cached and hence don't need timeout protection or should we be wrapping them too? Maybe none of this is a problem if we're moving to remote config anyways...